uint8 new_state